### PR TITLE
CI: Try fixing OSX

### PR DESCRIPTION
**Pull request by mensinda**

### CODE REVIEW

**Comment by xclaesse** [comment]
LGTM, once the CI confirms only NetCDF is still failing.

**Comment by mensinda** [comment]
Ideally, the NetCDF error should be fixed by @scivision with #6355. I don't know enough about Fortran and NetCDF to do a proper fix here.

**Comment by xclaesse** [comment]
I think we should just completely skip that test until @scivision figure it out. That test failed on his PR too.

**Comment by xclaesse** [comment]
+1 once the CI is green. Fingers crossed. Thanks!

**Comment by scivision** [comment]
I've been trying to have the scientific computing dependencies like NetCDF, HDF5 and similar tested only on Linux for Travis. And then testing the scientific computing dependencies for MacOS / Windows on GitHub Actions CI. In part to avoid overwhelming the Azure and Travis CI with too many dependency corner cases. GitHub Actions allows much finer-grained control of test setups and the MacOS on Actions seems more reliable than Travis at least to me.

So yes let's get a green light on the main Travis / Azure CI and the tricky project test cases I'll try to handle in a fine-grained way on Actions where it shouldn't bother the main CI project tests.

**Comment by scivision** [comment]
tldr; because of the rolling release nature of the scientific computing dependencies I'm interested in, it can be better to keep the trickier tests on Actions CI instead of Travis/Azure. What I understand from discussion with Jussi is to do a simple, stable dependency test on Travis with Linux, and then I can get the trickier cases on Actions, where it's much easier to diagnose or disable.

**Comment by jpakkane** [comment]
Linux things should be tested on the basic travis image as much as possible, because those tests are used on other Linux cases as well. For example, the Debian packaging and autopkgtests run the same test suite. The more things we have in one place, the easier they are to run under different scenarios.

**Comment by mensinda** [comment]
The Travis CI is already green. Should we wait for Azure before merging this?

**Comment by xclaesse** [comment]
Merged right away, because we have plenty of PRs to retry now.
